xdsl/dialects/pdl.py

```diff
@@ -34,10 +34,44 @@
     irdl_op_definition,
     IRDLOperation,
 )
+from xdsl.parser import Parser
+from xdsl.printer import Printer
 from xdsl.utils.exceptions import VerifyException
 from xdsl.utils.hints import isa
 
 
+def parse_operands_with_types(parser: Parser) -> list[SSAValue]:
+    """
+    Parse a list of operands with types of the following format:
+    `operand1, operand2 : type1, type2`
+    At least one operand is expected.
+    """
+    pos = parser.pos
+    operands = parser.parse_comma_separated_list(
+        Parser.Delimiter.NONE, parser.parse_operand
+    )
+    parser.parse_punctuation(":")
+    types = parser.parse_comma_separated_list(
+        Parser.Delimiter.NONE, parser.parse_attribute
+    )
+    end_pos = parser.pos
+    if len(operands) != len(types):
+        parser.raise_error(
+            "Mismatched between the numbers of operands and types", pos, end_pos
+        )
+    for operand, type in zip(operands, types):
+        if operand.typ != type:
+            raise Exception("Mismatched between operands and their types", pos, end_pos)
```

Comment:
Seems like this should be a subtype, why not `parser.raise_error` here also?

Reply:
Thanks! I changed it to `raise_error`.